application/src/main/java/org/thingsboard/server/actors/device/DeviceActorMessageProcessor.java

```diff
@@ -196,6 +208,22 @@ void processRpcRequest(TbActorCtx context, ToDeviceRpcRequestActorMsg msg) {
             syncSessionSet.forEach(rpcSubscriptions::remove);
         }
 
+        if (persisted) {
+            Rpc rpc = new Rpc(new RpcId(request.getId()));
+            rpc.setCreatedTime(System.currentTimeMillis());
+            rpc.setTenantId(tenantId);
+            rpc.setDeviceId(deviceId);
+            rpc.setExpirationTime(request.getExpirationTime());
+            rpc.setRequest(JacksonUtil.valueToTree(request));
+            rpc.setStatus(sent ? RpcStatus.SENT : RpcStatus.QUEUED);
```

Comment:
This is not 100% correct. You may send the notification to the session, but it will not be processed. You should save it as QUEUED and change to SENT only when you receive configuration from the transport level.

Reply:
Remote SENT and save tp the DB before you push to device session.